hphp/runtime/vm/jit/vasm-simplify-arm.cpp

```diff
@@ -84,17 +84,38 @@ bool simplify(Env& env, const cmovq& inst, Vlabel b, size_t i) {
 
 ///////////////////////////////////////////////////////////////////////////////
 
-bool simplify(Env& env, const loadb& inst, Vlabel b, size_t i) {
-  return if_inst<Vinstr::movzbl>(env, b, i + 1, [&] (const movzbl& mov) {
-    // loadb{s, tmp}; movzbl{tmp, d}; -> loadzbl{s, d};
-    if (!(env.use_counts[inst.d] == 1 &&
-          inst.d == mov.s)) return false;
+template<Vinstr::Opcode op>
+bool is_next(const Env& env, Vlabel b, size_t i)
+{
+  auto const& code = env.unit.blocks[b].code;
+  if (i > code.size()) return false;
+  return code[i].op == op;
+}
 
-    return simplify_impl(env, b, i, [&] (Vout& v) {
-      v << loadzbl{inst.s, mov.d};
-      return 2;
-    });
-  });
+bool simplify(Env& env, const loadb& inst, Vlabel b, size_t i) {
+  if (is_next<Vinstr::movzbl>(env, b, i + 1)) {
```

Comment:
Why add the new helper? Why not just write

  if (if_inst<...>(
    ...
  ) {
    return true;
  }
  return if_inst<...>(...);